### PR TITLE
[TT-13518] Updating OpenTelemetry version

### PR DESCRIPTION
**Pull request by mativm02**
<details open>
  <summary><a href="https://tyktech.atlassian.net/browse/TT-13518" title="TT-13518" target="_blank">TT-13518</a></summary>
  <br />
  <table>
    <tr>
      <th>Summary</th>
      <td>Otel e2e tests breaks after upgrading OTEL to v1.29</td>
    </tr>
    <tr>
      <th>Type</th>
      <td>
        Story
        Story
      </td>
    </tr>
    <tr>
      <th>Status</th>
      <td>In Dev</td>
    </tr>
    <tr>
      <th>Points</th>
      <td>N/A</td>
    </tr>
    <tr>
      <th>Labels</th>
      <td><a href="https://tyktech.atlassian.net/issues?jql=project%20%3D%20TT%20AND%20labels%20%3D%20'24Bugsmash%20ORDER%20BY%20created%20DESC" title="'24Bugsmash">'24Bugsmash</a></td>
    </tr>
  </table>
</details>

---

## Description
Updating OpenTelemetry version to get rid of a CVE.

## Related Issue
https://tyktech.atlassian.net/browse/TT-13518

## Motivation and Context
https://tyktech.atlassian.net/browse/TT-13518

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

**Comment by buger** [comment]
<p>
    Let's make that PR title a 💯 shall we? 💪
    </p>
    <p>
    Your <em>PR title</em> and <em>story title</em> look <strong>slightly different</strong>. Just checking in to know if it was intentional!
    </p>
    <table>
      <tr>
        <th>Story Title</th>
        <td>Otel e2e tests breaks after upgrading OTEL to v1.29</td>
      </tr>
      <tr>
          <th>PR Title</th>
          <td>[Updating OpenTelemetry version</td>
        </tr>
    </table>
    <p>
      Check out this <a href="https://www.atlassian.com/blog/git/written-unwritten-guide-pull-requests">guide</a> to learn more about PR best-practices.
    </p>

File: internal/graphql/otel_graphql_engine_detailed_test.go [hunk 1]
```diff
@@ -480,6 +480,10 @@ func (s *spanWrapper) TracerProvider() trace.TracerProvider {
 	return s.span.TracerProvider()
 }
 
+func (s *spanWrapper) AddLink(link trace.Link) {
+	s.span.AddLink(link)
+}
+
```
Comment (mativm02) [inline internal/graphql/otel_graphql_engine_detailed_test.go:486]: New method added by the library that must be implemented to satisfy the interface.

**Comment by buger** [comment]
<p>
    Knock Knock! 🔍
  </p>
  <p>
    Just thought I'd let you know that your <em>PR title</em> and <em>story title</em> look <strong>quite different</strong>. PR titles
    that closely resemble the story title make it easier for reviewers to understand the context of the PR.
  </p>
  <blockquote>
    An easy-to-understand PR title a day makes the reviewer review away! 😛⚡️
  </blockquote>
  <table>
    <tr>
      <th>Story Title</th>
      <td>Otel e2e tests breaks after upgrading OTEL to v1.29</td>
    </tr>
    <tr>
        <th>PR Title</th>
        <td>[TT-13518] Updating OpenTelemetry version</td>
      </tr>
  </table>
  <p>
    Check out this <a href="https://www.atlassian.com/blog/git/written-unwritten-guide-pull-requests">guide</a> to learn more about PR best-practices.
  </p>

**Comment by mativm02** [comment]
Latest OpenTelemetry's version already imported in https://github.com/TykTechnologies/tyk/pull/6727

**Pull request closed without merging**